src/Handler/Contract/CacheHandler.php

```diff
@@ -9,18 +9,20 @@ interface CacheHandler extends AnnotationHandler
     /**
      * @return mixed
      */
-    public function fetch(string $id);
+    public function fetch(string $poolName, string $id);
 
     /**
+     * @param array<int, string> $pools
      * @param array<int, string> $tags
      * @param mixed              $data
      */
-    public function save(string $id, $data, ?int $lifeTime = null, array $tags = []): void;
+    public function save(array $pools, string $id, $data, ?int $lifeTime = null, array $tags = []): void;
```

Comment:
here we should pass pool name, the interceptor should loop and call multiple saves